components/invitation_modal/invite_view.tsx

```diff
@@ -0,0 +1,339 @@
+    });
+    let noMatchMessageId = t(
+        'invitation_modal.members.users_emails_input.no_user_found_matching',
+    );
+    let noMatchMessageDefault =
+        'No one found matching **{text}**. Enter their email to invite them.';
+
+    if (!props.emailInvitationsEnabled) {
+        placeholder = formatMessage({
+            id: 'invitation_modal.members.search-and-add.placeholder-email-disabled',
+            defaultMessage: 'Add members',
+        });
+        noMatchMessageId = t(
+            'invitation_modal.members.users_emails_input.no_user_found_matching-email-disabled',
+        );
+        noMatchMessageDefault = 'No one found matching **{text}**';
+    }
+
+    const isInviteValid = useMemo(() => {
+        if (props.as === As.GUEST) {
+            return props.inviteChannels.channels.length > 0 && props.usersEmails.length > 0;
+        }
+        return props.usersEmails.length > 0;
+    }, [props.as, props.inviteChannels.channels, props.usersEmails]);
+
+    return (
+        <>
+            <Modal.Header className={props.headerClass}>
+                <h1>
+                    <FormattedMessage
+                        id='invite_modal.title'
+                        defaultMessage={'Invite {as} to {team_name}'}
+                        values={{
+                            as: (
+                                props.as === As.MEMBER ?
+                                    <FormattedMessage
+                                        id='invite_modal.members'
+                                        defaultMessage='members'
+                                    /> :
+                                    <FormattedMessage
+                                        id='invite_modal.guests'
+                                        defaultMessage='guests'
+                                    />
+                            ),
+                            team_name: props.currentTeam.display_name,
+                        }}
+                    />
+                </h1>
+                <button
+                    className='icon icon-close'
+                    aria-label='Close'
+                    title='Close'
+                    onClick={props.onClose}
+                />
+            </Modal.Header>
+            <Modal.Body>
+                <div className='InviteView__sectionTitle'>
+                    <FormattedMessage
+                        id='invite_modal.to'
+                        defaultMessage='To:'
+                    />
+                </div>
+                <UsersEmailsInput
+                    {...errorProperties}
+                    usersLoader={(value: string, cb: (users: UserProfile[]) => void) => {
+                        if (value === '') {
+                            clearMinHeight();
+                        }
+                        return props.usersLoader(value, cb);
+                    }}
+                    onUsersLoad={ensureInvitesVisible}
```

Comment:
By invoking this once the data is loaded, it should avoid the problem of failing to adjust size because of slow networks.